openedx/openedx-platform
# fix: thread will only be returned for matching course ID

## Description

Thread API was returning thread data even for the course the thread is not part of this has been handled in this PR and related frontend change has also been added in a separate PR for discussion MFE

## Supporting information

https://2u-internal.atlassian.net/browse/INF-669
https://github.com/openedx/frontend-app-discussions/pull/360

## Testing instructions

copy a thread Id from a valid course and append it in the URL of some other course that thread is not part of, the API should return 404 in this case.
 http://localhost:18000/api/discussion/v1/threads/{some_valid_thread_id}/?requested_fields=profile_image&course_id={some_valid_but_unrelated_course_id}

make sure to test the positive use case too
## Deadline
